### PR TITLE
map: scrolling over an images group scrolls the images of the group

### PR DESCRIPTION
**Pull request by phweyland**
some related changes:
- scrolling over an images group scrolls the images of the group
- fixes dnd of a group of images (only the first was moved since the grouping function)
- dnd of a group shows the next image while the first one is dragged

### CODE REVIEW

**Comment by phweyland** [comment]
Good for review.

**Comment by TurboGit** [comment]
Not sure it is related to this PR, but I have set the minimum of image for a group to 5. But only after having dnd 6 images close to each others I had a group created.

>     * fixes dnd of a group of images (only the first was moved since the grouping function)

Still the case for me.

BTW, another enhancement that could be great is to show the locations (circle or square) when moving them. It is hard otherwise to place it at the proper location.

File: src/views/map.c [hunk 1]
```diff
@@ -38,13 +38,22 @@
 
 DT_MODULE(1)
 
+typedef struct geo_position_t
```
Comment (TurboGit) [inline src/views/map.c:41]: dt_get_position_t

File: src/views/map.c [hunk 2]
```diff
@@ -38,13 +38,22 @@
...
+{
+  double x, y;
+  int cluster_id;
+  int imgid;
+} geo_position_t;
```
Comment (TurboGit) [inline src/views/map.c:46]: likewise

**Review by TurboGit — CHANGES_REQUESTED** [review]
Code looks ok, just a minor comment.

File: src/views/map.c (hunk 2 above)
Comment (phweyland) [reply src/views/map.c]: Done!

**Comment by phweyland** [comment]
> Not sure it is related to this PR, but I have set the minimum of image for a group to 5. But only after having dnd 6 images close to each others I had a group created.

Hummm. I've already checked this issue but I haven't found yet why. I don't give up.



> > ```
> > * fixes dnd of a group of images (only the first was moved since the grouping function)
> > ```
> 
> 
> Still the case for me.

It works well for me. Do you use shift-click ? Do you release the shift before the left click ?




> BTW, another enhancement that could be great is to show the locations (circle or square) when moving them. It is hard otherwise to place it at the proper location.

The flying pixbuf has no transparency. This seems to be a known issue of cairo pixbuf. So I've let the cross but it is really not ideal.

**Comment by phweyland** [comment]
> The flying pixbuf has no transparency. This seems to be a known issue of cairo pixbuf. So I've let the cross but it is really not ideal.

If you have any idea to solve this let me know. The dnd from filmstrip to map suffers the same issue.

**Comment by TurboGit** [comment]
> It works well for me. Do you use shift-click ? Do you release the shift before the left click ?

Oh I see, you need shift-click and you need to release shift before ! Not really nice :( Could all this be just a shift-click and whatever is done when releasing ?

**Comment by phweyland** [comment]
> > It works well for me. Do you use shift-click ? Do you release the shift before the left click ?
> 
> Oh I see, you need shift-click and you need to release shift before ! Not really nice :( Could all this be just a shift-click and whatever is done when releasing ?

Also an old behaviour. I look at it and tell you.

**Comment by phweyland** [comment]
Just remains the pixbuf transparency.

**Comment by phweyland** [comment]
I've found a way to control transparency differently (I don't know if that would work) but I cannot compile it:

      GdkPixmap *pixmap;
      GdkBitmap *mask;
      gdk_pixbuf_render_pixmap_and_mask(cross, &pixmap, &mask, 1);
      gtk_drag_set_icon_pixmap(context, NULL, pixmap, mask,
                               DT_PIXEL_APPLY_DPI(cross_size),
                               DT_PIXEL_APPLY_DPI(cross_size)); 

GdkPixmap, GdkBitmap, gdk_pixbuf_render_pixmap_and_mask and gtk_drag_set_icon_pixmap are unknown ...
Any idea on this ?

**Comment by parafin** [comment]
I'm not sure what exactly you are trying to achieve here, but you may want to check that your solution works on non-compositing window managers (see #5438 and #5382 for context).

**Comment by phweyland** [comment]
> I'm not sure what exactly you are trying to achieve here, but you may want to check that your solution works on non-compositing window managers (see #5438 and #5382 for context).

Thanks for jumping in.
Currently the icon (pixbuf) on drag and drop doesn't respect the alpha channel. This is not an issue for plain image but it is for an empty circle or rectangle.
I'll look at your links.

**Comment by phweyland** [comment]
@parafin, yes, css was the way to go. Thanks !

**Comment by phweyland** [comment]
Travis is happy. So I am.

**Comment by TurboGit** [comment]
@phweyland : the circle & rectangle when dragged are not adjusted for the ratio so a circle becomes an ellipse.

**Comment by TurboGit** [comment]
Apart from that, the shift-click and min number of image for creating a group is working fine on my side.

**Comment by phweyland** [comment]
> @phweyland : the circle & rectangle when dragged are not adjusted for the ratio so a circle becomes an ellipse.

I've also a slight difference between the original and the dragged one, but it is very small. 
Let me check.

**Comment by phweyland** [comment]
I've forgotten to check the most important, i.e. the Dieppe area, sorry ... :)
Well done. Thanks.

**Comment by TurboGit** [comment]
Ah ah ! Dieppe is THE place to check:) note that I was born there but I live around Paris now... Even if I would prefer not!

**Comment by TurboGit** [comment]
Perfect for me now, thanks !

**Comment by phweyland** [comment]
I've seen some discussions about docs. Where I should update them now ?

**Comment by TurboGit** [comment]
On the same documentation as before until 3.4. After that we will migrate to the new support.

**Pull request merged**